spec/math_spec.rb

```diff
@@ -1287,43 +1287,58 @@
 
           begin
             rank = a.svd_rank()
-          end
+          
+            rank_true = 3
+            expect(rank).to eq (rank_true)
 
-          rank_true = 3
-          expect(rank).to eq (rank_true)
+          rescue NotImplementedError
```

Comment:
Okay, interesting approach here. Generally if a method isn't supported under a user's specific configuration, we don't use rescue in specs. We just check with an `if` statement and use `pending` as you've done; we just do it without the `rescue`. Can you look around a bit and make it consistent with how we've done it elsewhere?